src/plugins/kibana_usage_collection/server/collectors/application_usage/rollups.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ISavedObjectsRepository, SavedObject, Logger } from 'kibana/server';
+import moment from 'moment';
+import {
+  ApplicationUsageDaily,
+  ApplicationUsageTotal,
+  ApplicationUsageTransactional,
+  SAVED_OBJECTS_DAILY_TYPE,
+  SAVED_OBJECTS_TOTAL_TYPE,
+  SAVED_OBJECTS_TRANSACTIONAL_TYPE,
+} from './saved_objects_types';
+import { SavedObjectsErrorHelpers } from '../../../../../../src/core/server';
+
+/**
+ * For Rolling the daily data, we only care about the stored attributes and the version (to avoid overwriting via concurrent requests)
+ */
+type ApplicationUsageDailyWithVersion = Pick<
+  SavedObject<ApplicationUsageDaily>,
+  'version' | 'attributes'
+>;
+
+/**
+ * Aggregates all the transactional events into daily aggregates
+ * @param logger
+ * @param savedObjectsClient
+ */
+export async function rollDailyData(logger: Logger, savedObjectsClient?: ISavedObjectsRepository) {
+  if (!savedObjectsClient) {
+    return;
+  }
+
+  try {
+    let toCreate: Map<string, ApplicationUsageDailyWithVersion>;
+    do {
+      toCreate = new Map();
+      const { saved_objects: rawApplicationUsageTransactional } = await savedObjectsClient.find<
+        ApplicationUsageTransactional
+      >({
+        type: SAVED_OBJECTS_TRANSACTIONAL_TYPE,
+        perPage: 1000, // Process 1000 at a time as a compromise of speed and overload
```

Comment:
We're specifying `perPage` for paginated results but not providing a `page` param. Looking at the [docs](https://www.elastic.co/guide/en/kibana/master/saved-objects-api-find.html#saved-objects-api-find-query-params), both are optional but how do we know which page we're going through? The docs also give a warning about using `find` for a large number of documents but don't say how many is `large'.